crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -541,11 +484,21 @@ impl Paginated<Cursor> for StoredTransaction {
     type Source = transactions::table;
 
     fn filter_ge<ST, GB>(cursor: &Cursor, query: Query<ST, GB>) -> Query<ST, GB> {
-        query.filter(transactions::dsl::tx_sequence_number.ge(cursor.tx_sequence_number as i64))
+        query
+            .filter(transactions::dsl::tx_sequence_number.ge(cursor.tx_sequence_number as i64))
+            .filter(
+                transactions::dsl::checkpoint_sequence_number
+                    .ge(cursor.tx_checkpoint_number as i64),
```

Comment:
Clarification question for my own understanding: the second filter on checkpoint number is actually not necessary since all txns satisfying the first filter would also satisfy the second, and we only include the second filter so that the index on checkpoint number can be hit. Is this correct?

Reply:
yes to both!